staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go

```diff
@@ -112,11 +112,31 @@ func newTestCacher(s storage.Interface, cap int) (*cacherstorage.Cacher, storage
 		NewFunc:        func() runtime.Object { return &example.Pod{} },
 		NewListFunc:    func() runtime.Object { return &example.PodList{} },
 		Codec:          codecs.LegacyCodec(examplev1.SchemeGroupVersion),
+		Clock:          clock.RealClock{},
 	}
 	cacher, err := cacherstorage.NewCacherFromConfig(config)
 	return cacher, v, err
 }
 
+func newTestCacherWithFakeClock(s storage.Interface) (*cacherstorage.Cacher, storage.Versioner, *clock.FakeClock, error) {
```

Comment:
Please avoid code duplication.
Let's make

newTestCacherWithClock(s *storage.Interface, clock *clock.Clock) (...) {
  // reuse function above just setting the clock appropriately
}

newTestCacher(s storage.Interface) (...) {
  return newTestCacherWithClock(s, clock.RealClock{})
}